### DISCUSSION TITLE
query using association's column(s)

### DISCUSSION THREAD

**Discussion by zgid123** [mikro-orm/mikro-orm#4496] [Q&A] [answered]
Hi,

I have three models: A, B and C. A has many B, and B has many A via C.

```ts
@Entity()
class A {
  @OneToMany(() => C, (c) => c.a)
  cs = new Collection<C>(this);
}

class C {
  @ManyToOne(() => B)
  b: B;

  @ManyToOne(() => A)
  a: A;
}

class B {
  @Property({
    nullable: false,
  })
  slug: string;

  @OneToMany(
    () => C,
    (c) => c.b,
  )
  cs = new Collection<C>(this);
}
```

One of my project used `mikro-orm` version `6.0.0-dev.15`, I do query like this one

```ts
this.aRepository.find({
  cs: {
    b: {
      slug,
    }
  }
});
```

But after upgrading to `6.0.0-dev.87`, it raises error

```
Type '{ b: { slug: string; }; }' is not assignable to type 'Collection<C, object>'
```

So I wonder how to do this correctly in new version?

**Comment by B4nan** [comment, marked as answer]
It works fine for me, nothing changed about this.

https://stackblitz.com/edit/mikro-orm-v6-repro-ltquhy?file=basic.test.ts

**Reply by zgid123** [reply]
oh, I just found out, when I use like this, it will raise error

```ts
let query: FilterQuery<A> = {};

if (bSlug) {
  query = {
    ...query,
    cs: {
      b: { slug: bSlug },
    },
  };
}
```

this will raise error, I don't know why. So I think I will fix using `Object.assign`

Thank you for your rely
